src/Roslyn.Suppressions/Roslyn.Suppressions/SupportedSuppressionBuilder.g.cs

```diff
@@ -0,0 +1,484 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using System;
+using System.Collections.Immutable;
+
+namespace SonarLint.VisualStudio.Roslyn.Suppressions
+{
+    /// <summary>
+    /// Generated class that returns SupportedSuppressions for all Sonar C# and VB.NET rules
+    /// </summary>
+    internal sealed class SupportedSuppressionsBuilder
+    {
+        private static readonly Lazy<SupportedSuppressionsBuilder> lazy = new Lazy<SupportedSuppressionsBuilder>(() => new SupportedSuppressionsBuilder());
+
+        public static SupportedSuppressionsBuilder Instance => lazy.Value;
+
+        public ImmutableArray<SuppressionDescriptor> Descriptors { get; }
+
+        private SupportedSuppressionsBuilder()
+        {
+            Descriptors = GetDescriptors();
+        }
+
+        private static ImmutableArray<SuppressionDescriptor> GetDescriptors()
+        {
+            var descriptors = new SuppressionDescriptor[]
+            {
+                // Number of diagnostics (C# and VB.NET): 429
```

Comment:
Do we have a way to know if this number is indeed the number of diagnostics this version of the analyzer has?

Reply:
Not really. We're looking at two different analysers with differing numbers of rules, where some of the rule ids overlap.
This is just the number of unique ids across the two analyzers.

It's there mainly for information, and as a warning sign: if we update the analysers and the number of rules falls, we should probably investigate further.